src/Plugin/resource/DataProvider/DataProviderEntity.php

```diff
@@ -592,28 +598,73 @@ protected function queryForListSort(\EntityFieldQuery $query) {
    */
   protected function queryForListFilter(\EntityFieldQuery $query) {
     $resource_fields = $this->fieldDefinitions;
-    foreach ($this->parseRequestForListFilter() as $filter) {
+    $filters = $this->parseRequestForListFilter();
+    $this->validateFilters($filters);
+    foreach ($filters as $filter) {
       // Determine if filtering is by field or property.
       /* @var ResourceFieldEntityInterface $resource_field */
       if (!$resource_field = $resource_fields->get($filter['public_field'])) {
-        return;
+        if (!static::isNestedField($filter['public_field'])) {
+          // This is not a nested filter.
+          continue;
+        }
+        if (!empty($filter['target'])) {
+          // If we cannot find the field, it may be a nested filter. Check if
+          // the target of that is the current resource.
+          continue;
+        }
+        $this->addNestedFilter($filter, $query);
+        continue;
       }
       if (!$property_name = $resource_field->getProperty()) {
         throw new BadRequestException(sprintf('The current filter "%s" selection does not map to any entity property or Field API field.', $filter['public_field']));
       }
+
       if (field_info_field($property_name)) {
         if (in_array(strtoupper($filter['operator'][0]), array('IN', 'BETWEEN'))) {
-          $query->fieldCondition($property_name, $resource_field->getColumn(), $filter['value'], $filter['operator'][0]);
+          $query->fieldCondition($property_name, $resource_field->getColumn(), $this->getReferencedIds($filter['value'], $resource_field), $filter['operator'][0]);
           continue;
         }
         for ($index = 0; $index < count($filter['value']); $index++) {
-          $query->fieldCondition($property_name, $resource_field->getColumn(), $filter['value'][$index], $filter['operator'][$index]);
+          // If referencing an entity by an alternate ID, change the user
+          // provided value by the entity value.
```

Comment:
"by the entity value" - what do you mean? If user gave a `name`, what do we query here - the ID?